### DISCUSSION TITLE
newTreeshaking cause build time from 10s to 1.4min

### DISCUSSION THREAD

**Discussion by jayjliang** [web-infra-dev/rspack#6042] [Q&A] [answered]
Our project has thounds of dependencies and most of them are private, so i can not provide the demo.but i use [Mac Xcode Instruments](https://web-infra-dev.github.io/rspack-dev-guide/profiling/intro.html#mac-xcode-instruments) to get profiling data
<img width="1591" alt="image" src="https://github.com/web-infra-dev/rspack/assets/9363437/21a7788a-56cb-4f54-a692-44b32ced2e7f">

so is there anyone who can help me to solve this problem

**Comment by jayjliang** [comment]
https://github.com/jayjliang/data

profiling data link

**Comment by LingyuCoder** [comment]
The XCode trace shows that there is a lot of idle time on the Rust side, so this issue may be caused by the logic on the JavaScript side.

You can generate the trace files of both Rust and JavaScript by using `RSPACK_PROFILE=ALL rspack build`([more detail](https://www.rspack.dev/guide/profile.html)).

**Reply by jayjliang** [reply]
[rspack-profile-1711970057544.zip](https://github.com/web-infra-dev/rspack/files/14822902/rspack-profile-1711970057544.zip)

**Reply by jayjliang** [reply]
it seems that js side is also normal

**Comment by IWANABETHATGUY** [comment]
can you try 
0.5.9-canary-c269a3b-20240401072545

**Reply by jayjliang** [reply]
it still cost 63s for build, if turn off newTreeshaking, the build time is 12s

**Reply by jayjliang** [reply]
@IWANABETHATGUY should I provide the new profile data to solve this problem?

**Reply by IWANABETHATGUY** [reply]
Can you try  the latest nightly?

**Reply by jayjliang** [reply]
no problem, but where to get the latest nightly version?

**Reply by xc2** [reply]
```sh
npm info @rspack/core@nightly version
```

**Reply by jayjliang** [reply]
the latest nightly version `0.6.1-canary-73a9832-20240414004901` works! Build time reduce to 10s, thanks，guys

by the way, which version of rsbuild will has this great job

**Reply by xc2** [reply, marked as answer]
Give a try to rsbuild 0.6.2 or rspack 0.6.1

**Reply by jayjliang** [reply]
rsbuild 0.6.2 works well~ thanks，guys
